### PR TITLE
Ember data model lookups

### PR DESCRIPTION
**Pull request by fivetanley**
This pull request will transform hasMany/belongsTo lookups to use the dasherized string form of the lookup. Looking up via camelCase, variable, or `App.Post` will be removed for Ember Data 1.0.

I think we can do better here by adding this transformation to store methods as well.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,6 +38,37 @@ want to convert code somewhere different to `app/`.
...
+This changes the way model lookups happen when working with Ember
+Data. When using `hasMany` or `belongsTo`, the first argument will
+become a camelized string. Here's some examples:
+
+```javascript
+
+// before, using a camelCase string
+
+export default DS.Model.extend({
+  postComments: DS.hasMany('post-comment', {async: true})
```
Comment (kamal) [inline README.md]: this should be `'postComment'`

**Comment by abuiles** [comment]
@fivetanley  thanks! I'll have a second look tomorrow morning and release :), seems  like @kamal's comment is a simple typo, right?

File: lib/formulas/ember-data-model-lookup-transform.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      if (isRelationshipMacro(node)) {
+        replaceArguments(path);
+      }
+
+      this.traverse(path);
+    }
+  });
+
+  function isHasMany(node) {
+    if (types.MemberExpression.check(node.callee)) {
```
Comment (abuiles) [inline lib/formulas/ember-data-model-lookup-transform.js]: :+1:  awesome! I missed this built-in checks.

File: README.md (hunk 1 above)
Comment (fivetanley) [inline README.md]: ah, totally. thanks!

**Comment by abuiles** [comment]
@fivetanley  thanks!

**Pull request merged**